### DISCUSSION TITLE
Trino with HA

### DISCUSSION THREAD

**Discussion by Utkarsh-0211** [trinodb/trino#13910] [Q&A]
I have deployed Trino coordinator and worker as stateful set in HA mode

Problem statement : I want to restart the pods as and when I add new catalogue in my config map so that it is updated.

Solution tried :
1. Restarted the stateful set directly so the 2 coordinator and 2 worker deployed get restarted and the config map is updated.But the problem is I'm not able to identify which one is the primary coordinator/worker and which one is the second. Basis the sequencing yes it is identifiable but once the primary is down and the secondary takes it's place and the indexing goes for a toss.

Desired Solution : 
Would like a solution wherein the instead of restarting the stateful set I restart the secondary coordinator/worker pod first and then the primary one. Any way to give an identity to it in the kubectl restart command ?

**Comment by Utkarsh-0211** [comment]
@dain @ebyhr @tangjiangling @findepi
@martint
Request for your kind help please.

**Comment by hashhar** [comment]
See https://github.com/trinodb/trino/issues/12709 which will bring such capability into engine. Anything else for now is a hack.

> restart the secondary coordinator/worker pod first and then the primary one. Any way to give an identity to it in the kubectl restart command

Seems like a kubernetes question and you might have better luck on the #kubernetes channel on Trino slack or a more k8s focused place. But to the best of my knowledge it's not possible to differentiate between "pods" within a set.
